engine/src/php5/_value.c

```diff
@@ -31,9 +31,20 @@ static int _value_current_key_get(HashTable *ht, char **str_index, ulong *num_in
 
 static void _value_current_key_set(HashTable *ht, engine_value *val) {
 	zval *tmp;
+	char *key;
+	unsigned long index;
 
 	MAKE_STD_ZVAL(tmp);
-	zend_hash_get_current_key_zval(ht, tmp);
```

Comment:
Also, the addition of this function was made in relation to the fact that you can have [non-scalar keys as iterators in foreach](https://wiki.php.net/rfc/foreach-non-scalar-keys) which obviously this doesn't handle. Not sure if this is a use-case we want to handle, but it's a question of what the lowest common denominator is.